dmci/api/app.py

```diff
@@ -206,33 +206,44 @@ def _distributor_wrapper(self, worker):
         return err
 
     @staticmethod
-    def _check_namespace_UUID(metadata_id, env_string=None):
-        """Splits incoming metadata_id to namespace and UUID, assuming
-        structure to be namespace:UUID
+    def _check_metadata_id(metadata_id, env_string=None):
+        """ Check that the metadata_id is structured as
+        namespace:UUID, that the uuid part is of type UUID, and that
+        the namespace is correct.
         """
-        md_uuid, md_namespace, err = None, "", None
-        elements = metadata_id.split(":")
+        # Initialize variables
+        md_namespace, md_uuid, err = None, None, None
+
+        # Split metadata_id in namespace and uuid parts and, as such,
+        # also check the stucture
         try:
-            md_uuid = uuid.UUID(elements[-1])
-            # Only UUID, no namespace
-            if len(elements) < 2:
-                md_namespace, err = "", None
-            # namespace and UUID
-            elif len(elements) == 2:
-                md_namespace, err = elements[0], None
-            else:
-                err = "Malformed metadata id. Should be <namespace>:<UUID>."
-                logger.error(err)
-                return "", None, err
+            md_namespace, uuid_str = metadata_id.split(":")
+        except ValueError as e:
+            # E.g., if namespace is missing, or there are more than
+            # one colon
+            logger.error(str(e))
+            return None, None, f"Input must be structured as <namespace>:<uuid>."
 
+        # Check if the provided uuid is a valid UUID.
+        try:
+            md_uuid = uuid.UUID(uuid_str)
         except ValueError as e:
-            err = f"Can not convert to UUID: {elements[-1]}"
-            logger.error(err)
             logger.error(str(e))
+            return None, None, f"Cannot convert to UUID: {uuid_str}"
 
+        # Check that the namespace is correct
         if env_string is not None:
-            if env_string != md_namespace.split(".")[-1]:
-                md_namespace = md_namespace + "." + env_string
+            env = md_namespace.split(".")[-1]
+            if env_string != env:
+                return None, None, f"Dataset metadata_id namespace is wrong: {env}"
```

Comment:
Should or should not the end-user have to know which env he posts to?

Reply:
We do not check and return error in /create /insert /update

Reply:
no, but delete is different. See the example in the issue (#171). It is very easy to make a mistake, and delete in the wrong environment, and then it becomes all confusing and stressful.. I've done it several times already..|